src/Compilers/CSharp/Test/Emit2/FlowAnalysis/RegionAnalysisTests.cs

```diff
@@ -12358,5 +12358,1743 @@ static void verify(DataFlowAnalysis analysis)
+        [CombinatorialData]
+        public void InlineArrays_064(bool isRef, [CombinatorialValues(0, 1)] int index)
+        {
+            DataFlowAnalysis analysis = CompileAndAnalyzeDataFlowExpression(
+@"
+class B
+{
+    static void M(" + (isRef ? "ref " : "") + @"Buffer22 x)
+    {
+        ref int y = ref x[" + index + @"][/*<bind>*/" + index + @"/*</bind>*/];
+        y++;
+    }
+}
+" + InlineArray22Definition, TargetFramework.Net80);
+
+            Assert.Null(GetSymbolNamesJoined(analysis.AlwaysAssigned));
+            Assert.Null(GetSymbolNamesJoined(analysis.Captured));
+            Assert.Null(GetSymbolNamesJoined(analysis.CapturedInside));
+            Assert.Null(GetSymbolNamesJoined(analysis.CapturedOutside));
+            Assert.Null(GetSymbolNamesJoined(analysis.VariablesDeclared));
+
+            Assert.Null(GetSymbolNamesJoined(analysis.DataFlowsIn));
+            Assert.Null(GetSymbolNamesJoined(analysis.DataFlowsOut));
+
+            Assert.Equal("x", GetSymbolNamesJoined(analysis.DefinitelyAssignedOnEntry));
+            Assert.Equal("x", GetSymbolNamesJoined(analysis.DefinitelyAssignedOnExit));
+
+            Assert.Null(GetSymbolNamesJoined(analysis.ReadInside));
+            Assert.Equal("x, y", GetSymbolNamesJoined(analysis.ReadOutside));
+
+            Assert.Null(GetSymbolNamesJoined(analysis.WrittenInside));
+            Assert.Equal("x, y", GetSymbolNamesJoined(analysis.WrittenOutside));
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void InlineArrays_070([CombinatorialValues("ref ", "in ", "")] string refModifier, [CombinatorialValues(0, 1)] int index)
+        {
+            DataFlowAnalysis analysis = CompileAndAnalyzeDataFlowExpression(
+@"
+class B
+{
+    static void M(" + refModifier + @"Buffer2 x)
+    {
+        ref readonly int y = ref /*<bind>*/x[" + index + @"]/*</bind>*/;
+        _ = y;
+    }
+}
+" + InlineArray2Definition, TargetFramework.Net80);
+
+            Assert.Null(GetSymbolNamesJoined(analysis.AlwaysAssigned));
+            Assert.Null(GetSymbolNamesJoined(analysis.Captured));
+            Assert.Null(GetSymbolNamesJoined(analysis.CapturedInside));
+            Assert.Null(GetSymbolNamesJoined(analysis.CapturedOutside));
+            Assert.Null(GetSymbolNamesJoined(analysis.VariablesDeclared));
+
+            Assert.Equal("x", GetSymbolNamesJoined(analysis.DataFlowsIn));
+            Assert.Null(GetSymbolNamesJoined(analysis.DataFlowsOut));
+
+            Assert.Equal("x", GetSymbolNamesJoined(analysis.DefinitelyAssignedOnEntry));
+            Assert.Equal("x", GetSymbolNamesJoined(analysis.DefinitelyAssignedOnExit));
+
+            Assert.Equal("x", GetSymbolNamesJoined(analysis.ReadInside));
+            Assert.Equal(refModifier != "" ? "x, y" : "y", GetSymbolNamesJoined(analysis.ReadOutside));
+
+            Assert.Equal("x", GetSymbolNamesJoined(analysis.WrittenInside));
+            Assert.Equal("x, y", GetSymbolNamesJoined(analysis.WrittenOutside));
```

Comment:
>x [](http://example.com/codeflow?start=26&length=1)

Why is `x` considered written (inside or outside)?

Reply:
> Why is x considered written (inside or outside)? Is it because a reference was taken to the contents of x, regardless of whether that reference was written through? Or because it is a parameter that is not out?

The latter: "because it is a parameter that is not out".